### PR TITLE
Add the ability for users to view their API keys

### PR DESCRIPTION
**Pull request by mjriley**
## Product Description
For users that have a managing IDP configured with the `always_show_user_api_keys` property, they will now be able to copy their existing API keys. In addition, some of the wording has been changed around storing your key. Finally, expiration dates for IDPs with a maximum expiration date now have better support in the calendar widget.
Screenshot 2024-03-25 at 6 37 06 PM

## Technical Summary
Associated ticket: https://dimagi.atlassian.net/browse/SAAS-15362

## Feature Flag
This display does not technically require a feature flag, but the `MULTI_VIEW_API_KEYS` flag is required to configure an IDP with the properties needed to display these fields.

### Safety story
Most of this functionality is essentially locked behind a feature flag. All functionality was tested locally and through unit tests

### Automated test coverage

New test suites were created in _settings/tests/test_views_ and _settings/tests/test_forms_

### QA Plan

QA will be done with the full feature

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/settings/forms.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 from django import forms
 from django.contrib.auth.forms import PasswordChangeForm
 from django.contrib.postgres.forms import SimpleArrayField
+from django.core.exceptions import ValidationError
 from django.utils.html import format_html
+from django.utils import timezone
 from django.utils.translation import gettext as _
 from django.utils.translation import gettext_lazy
+import pytz
```
[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/settings/forms.py:11]: Interesting, was definitely not aware of this. I'll take a look at how best to grab timezones now.

[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/settings/forms.py]: Moved over to `ZoneInfo`: c5b115cb56925b93175044bde56b1577fd0923ee

File: corehq/apps/settings/forms.py [hunk 2]
```diff
@@ -272,13 +276,29 @@ class HQApiKeyForm(forms.Form):
...
+
+        if max_allowed_expiration_days is not None:
+
+            self.fields['expiration_date'].required = max_allowed_expiration_days is not None
+            today = datetime.now(tz=self.timezone)
+            self.maximum_expiration_date = today + timedelta(days=max_allowed_expiration_days)
+
+            num_days_displayed = self.get_expiration_window_display_string(max_allowed_expiration_days)
+            appended = _('Must be no later than {} from today: {}').format(
+                num_days_displayed.lower(), self.maximum_expiration_date.strftime(USER_DATE_FORMAT)
```
[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/settings/forms.py:297]: Addressed in 9ac233935281c09f5218113ee6fff289f26b9bec

File: corehq/apps/settings/forms.py [hunk 3]
```diff
@@ -346,7 +346,7 @@ def clean_expiration_date(self):
         if not self.cleaned_data['expiration_date']:
             return self.cleaned_data['expiration_date']
 
-        expiration_date = self.timezone.localize(self.cleaned_data['expiration_date'])
+        expiration_date = self.cleaned_data['expiration_date'].replace(tzinfo=self.timezone)
```
Comment (millerdev) [inline corehq/apps/settings/forms.py:351]: Does this do the same thing that `timezone.localize(...)` did? I think you might want to use [`datetime.astimezone`](https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone).

Comment (mjriley) [reply corehq/apps/settings/forms.py:351]: Yes, localize worked the same way -- it took a naive time and just tacked the given timezone to it. `replace` does the same thing, whereas `astimezone` assumes that naive object is UTC. I tested this locally, but when I'm using this widget, and I send across a date, I am sending that date across in my local time, and the time attached to it is '00:00:00' -- it is important that that does not first get adjusted to UTC

Comment (mjriley) [reply corehq/apps/settings/forms.py:351]: It's probably also worth noting that this specific case is covered in _test_forms:HQApiKeyTests.test_expiration_date_is_localized_

**Review by millerdev — APPROVED** [review]

**Pull request merged**